src/astprinter.nit

```diff
@@ -90,10 +175,66 @@ redef class ABlockExpr
 	end
 end
 
-redef class AIntegerExpr
-	redef fun accept_printer(v)
+redef class ADoExpr
+	redef fun number_of_lines do return n_block.number_of_lines
+	redef fun accept_printer(v) do
+		v.visit_block("do", n_block)
+	end
+end
+
+redef class ALoopExpr
+	redef fun number_of_lines do return n_block.number_of_lines
+	redef fun accept_printer(v) do
+		v.visit_block("loop", n_block)
+	end
+end
+
+redef class AForExpr
+	redef fun number_of_lines do return n_block.number_of_lines
+	redef fun accept_printer(v) do
+		v.write("for ")
+		for g in n_groups do
+			v.write(v.varname(g.variables.first))
+			v.write(" in ")
+			v.visit(g.n_expr)
+		end
+		v.visit_block("do", n_block)
+	end
+end
+
+redef class AIfExpr
+	redef fun number_of_lines
 	do
-		v.write(value.to_s)
+		 #n_then.number_of_lines
+		 #n_else.number_of_lines
+		 return 3
+	end
+	redef fun accept_printer(v) do
+		v.write("if ")
+		v.enter_visit(n_expr)
+		if n_else != null then
+			v.visit_block_noend("then", n_then)
+			v.visit_block("else", n_else)
+		else
+			v.visit_block("then", n_then)
+		end
+	end
+end
+
+redef class AIfexprExpr
+	redef fun number_of_lines
+	do
+		 #n_then.number_of_lines
+		 #n_else.number_of_lines
+		 return 3
+	end
+	redef fun accept_printer(v) do
+		v.write("if ")
+		v.enter_visit(n_expr)
+		v.write(" then ")
+		v.visit(n_then)
+		v.write(" else ")
+		v.visit(n_else)
```

Comment:
Conditionnal on the else existing?

Reply:
this is for the ternary operator, there is always an else part